tdrs-backend/scripts/db_backup.py

```diff
@@ -0,0 +1,228 @@
+
+    # write .pgpass
+    with open('/home/vcap/.pgpass', 'w') as f:
+        f.write(DATABASE_HOST+":"+DATABASE_PORT+":"+DATABASE_DB_NAME+":"+DATABASE_USERNAME+":"+DATABASE_PASSWORD)
+    os.environ['PGPASSFILE'] = '/home/vcap/.pgpass'
+
+except Exception as e:
+    print(e)
+    sys.exit(1)
+
+
+def backup_database(file_name,
+                    postgres_client,
+                    database_uri):
+    """Back up postgres database into file."""
+    """
+    :param file_name: back up file name
+    :param postgres_client: directory address for postgres application
+    :param database_uri: postgres URI
+    pg_dump -F c --no-acl --no-owner -f backup.pg postgresql://${USERNAME}:${PASSWORD}@${HOST}:${PORT}/${NAME}
+    """
+    try:
+        os.system(postgres_client + "pg_dump -Fc --no-acl -f " + file_name + " -d " + database_uri)
+        return True
+    except Exception as e:
+        logging.log(e)
+        return False
+
+
+def restore_database(file_name, postgres_client, database_uri=DATABASE_URI):
+    """Restore the database from filename."""
+    """
+    :param file_name: database backup filename
+    :param postgres_client: directory address for postgres application
+    """
+    try:
+        os.system(postgres_client + "createdb " + "-U " + DATABASE_USERNAME + " -h " + DATABASE_HOST +
+                  " -T template0 " + DATABASE_DB_NAME)
+    except Exception as e:
+        logging.log(e)
+    try:
+        os.system(postgres_client + "pg_restore --clean --no-owner --no-privileges --no-acl --create"
+                                    " -d " + database_uri + " " + file_name)
+        return True
+    except Exception as e:
+        logging.log(e)
+        return False
+
+
+def upload_file(file_name, bucket, object_name=None, region='us-gov-west-1'):
+    """Upload a file to an S3 bucket."""
+    """
+    :param file_name: file name being uploaded to s3 bucket
+    :param bucket: bucket name
+    :param object_name: S3 object name. If not specified then file_name is used
+    :param region: s3 AWS region to be used. defaults to government west
+    :return: True is file is uploaded, False if not successful
+    """
+    if object_name is None:
+        object_name = os.path.basename(file_name)
+    # upload the file
+    s3_client = boto3.client('s3', region_name=region)
+    try:
+        s3_client.upload_file(file_name, bucket, object_name)
+        return True
+    except Exception as e:
+        logging.error(e)
+        return False
+
+
+def download_file(bucket,
+                  file_name,
+                  region,
+                  object_name=None,
+                  ):
+    """Download file from s3 bucket."""
+    """
+    :param bucket
+    :param file_name
+    :param region
+    :param object_name
+    """
+    if object_name is None:
+        object_name = os.path.basename(file_name)
+    s3 = boto3.client('s3', region_name=region)
+    s3.download_file(bucket, object_name, file_name)
```

Comment:
Will this overwrite a pre-existing file?

Reply:
Yes it will. We should set a filename convention. E.g with date in the filename